### PR TITLE
Add PriceFeedInterface

### PR DESCRIPTION
**Pull request by mrice32**
This adds a simple interface that all price feeds should implement.

This PR also includes changes to force the UniswapPriceFeed to conform to this interface, and it moves the price feeds to their own directory.

### CODE REVIEW

File: ci/run_truffle_tests.sh [hunk 1]
```diff
@@ -47,7 +47,8 @@ run_tests $PROTOCOL_DIR/core
 
 # Hacky way of running truffle tests outside of core.
 cd $PROTOCOL_DIR/core
-$(npm bin)/truffle test ../financial-templates-lib/test/*.js --network ci
+
+$(npm bin)/truffle test $(find ../financial-templates-lib/test -name '*.js') --network ci
```
Comment (nicholaspai) [inline ci/run_truffle_tests.sh:51]: Would this script work for the other scaffolding directories `liquidator`, `disputer` and `monitors`?

File: financial-templates-lib/price-feed/PriceFeedInterface.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  // This returns the last time that the `update()` method was called. If it hasn't been calle
+  // Note: derived classes *must* override this method.
+  getLastUpdateTime() {
+    this._abstractFunctionCalled();
+  }
+
+  // Common function to throw an error if an interface method is called.
+  _abstractFunctionCalled() {
```
Comment (nicholaspai) [inline financial-templates-lib/price-feed/PriceFeedInterface.js:34]: I like this abstract class structure a lot

File: ci/run_truffle_tests.sh (hunk 1 above)
Comment (mrice32) [reply ci/run_truffle_tests.sh:51]: Yeah, the reason I only added it here is that this is the only one that has a nested directory. Do you think we should go ahead and do it for all of them?

Comment (mrice32) [reply ci/run_truffle_tests.sh:51]: Done.

**Pull request merged**